### DISCUSSION TITLE
Select File - Dialog is opening after multiple clicks

### DISCUSSION THREAD

**Discussion by vinodsharmak** [wailsapp/wails#972] [Q&A]
I am facing some issue with Select File dialog. I was on 1.16.7 version and select file dialog is opening on single click. I have to several times to open it

Code: runtime.Dialog.SelectFile()

Then I update the wails version to 1.16.8 and is working fine locally but after cross compiling it is not working. I have to click several times to open a file dialog. Please let mw know how can I resolve this issue.

I observed one thing on docker hub, version is 1.16.4 (https://hub.docker.com/r/wailsapp/xgo/tags). Please let me know how can this be resolved. Need docker image for 1.16.8

**Comment by leaanthony** [comment]
Hey there. Cross compilation is a bit broken for v1. If you search the issues, you'll see someone managed to get the updated xgo working bit nobody has done a PR yet.

**Reply by vinodsharmak** [reply]
Thanks. We will search issues

**Reply by vinodsharmak** [reply]
I did not find anything that could help. I tried with 1.16.3 to match with docker image 1.16.3 but no luck. Please suggest

**Reply by vinodsharmak** [reply]
Issue resolved. 

Non working code-
func (ft *FileTransfer) SelectFile() string {
	filename:= ft.runtime.Dialog.SelectFile()
        return filename
}

**Working code-
func (ft *FileTransfer) SelectFile() string {
	return ft.runtime.Dialog.SelectFile()
}**

**Note:-** one should not assign the value of SelectFile() to a variable
